### ISSUE TITLE
C++ constraints shall use double instead of int64_t when type is number

### ISSUE THREAD

**Issue by jfgo20** [glideapps/quicktype#2101]
For example, if we  have this in the schema:

```
"test": {
  "title": "Test",
  "type": "number",
  "minimum": -1.7e+308,
  "maximum": 1.7e+308
}
```

It generates this C++ code:

```
test_constraint(-1.7e+308, 1.7e+308, std::nullopt, std::nullopt, std::nullopt)
```

But ClassMemberConstraints is using int64_t for min_value/max_value:

```
class ClassMemberConstraints {
    private:
    std::optional<int64_t> min_value;
    std::optional<int64_t> max_value;
```
So, we should have 2 different classes for constraints, one for int64_t, and another for double.

**Comment by michadn** [comment]
@dvdsgl I think this issue can be closed due to the merged PR. Thanks for your work!

**Comment by inferrinizzard** [comment]
Closing as resolved in #2257
